lib/gpw/client.rb

```diff
@@ -2,36 +2,101 @@
 
 module GPW
   class Client
+    def initialize(client: Faraday.new)
+      @client = client
+    end
+
     def fetch_companies_list
-      response = fetch_gpw_stock_list
+      response = fetch_stock_list
       parse_response(response)
     end
 
-    private
+    def fetch_gpw_companies_details
+      response = fetch_gpw_stock_list
+      parse_money_response_gpw(response)
+    end
 
-    def fetch_gpw_stock_list
-      conn_gpw = Faraday.new(
-        url: 'https://gpw.notoria.pl/widgets/ta/symbols.php',
-        headers: {
-          'Accept' => 'application/json',
-          'Referer' => 'https://gpw.notoria.pl',
-          'DNT' => '1',
-          'Sec-GPC' => '1',
-        },
-      )
+    def fetch_nc_companies_details
+      response = fetch_nc_stock_list
+      parse_money_response_nc(response)
+    end
 
-      conn_gpw.get
+    private
+
+    def fetch_stock_list
+      @client.get do |req|
+        req.url 'https://gpw.notoria.pl/widgets/ta/symbols.php'
+        req.headers['Accept'] = 'application/json'
+        req.headers['Referer'] = 'https://gpw.notoria.pl'
+        req.headers['DNT'] = '1'
+      end
     end
 
     def parse_response(response)
       body_parsed = JSON.parse(response.body.split('(')[1].split(')')[0])
 
       body_parsed.fetch('symbols').map do |company|
+        next unless company[1]
```

Comment:
I will produce `nil`s in the resulting array right? Do you want them?

I think it would make more sense to just filter them out:

```ruby
body_parsed.fetch('symbols').reject { |company| company[1] }.map ...